inject/src/main/java/io/micronaut/context/AbstractInitializableBeanDefinition.java

```diff
@@ -177,9 +175,59 @@ protected AbstractInitializableBeanDefinition(
         this.precalculatedInfo = precalculatedInfo;
     }
 
-    @Override
-    public final boolean isConfigurationProperties() {
-        return precalculatedInfo.isConfigurationProperties;
+    /**
+     * Represents {@link BeanDefinitionReference#getBeanDefinitionName()} when the class implements {@link BeanDefinitionReference}.
+     *
+     * @return The name of this bean definition
+     */
+    // @Override for BeanDefinitionReference
+    public String getBeanDefinitionName() {
```

Comment:
i think it would be good to introduce another subclass of AbstractInitializableBeanDefinition that implements BeanDefinitionReference and properly overrides these methods. It is both more maintainable, and it may be better for performance.